com.microsoft.mrtk.tools/SubsystemWizard/SubsystemGenerator.cs

```diff
@@ -0,0 +1,595 @@
+
+            bool success = ValidateName(BaseClassName);
+
+            error = success ? string.Empty :
+                $"Subsystem base class name: {BaseClassName} is not a valid C# identifier.";
+
+            return success;
+        }
+
+        /// <summary>
+        /// Ensures that the value of <see cref="SubsystemNamespace"/> is valid for the
+        /// C# language.
+        /// </summary>
+        /// <param name="error">String describing the encountered error.</param>
+        /// <returns>True of successful, or false.</returns>
+        public bool ValidateNamespace(out string error)
+        {
+            // Ensure a name was provided.
+            if (string.IsNullOrWhiteSpace(SubsystemNamespace))
+            {
+                error = $"Subsystem namespace: Must specify a name.";
+                return false;
+            }
+
+            bool success = true;
+            StringBuilder sb = new StringBuilder();
+
+            string[] namespaceComponents = SubsystemNamespace.Split('.');
+
+            foreach (string s in namespaceComponents)
+            {
+                if (!ValidateName(s))
+                {
+                    if (sb.Length > 0)
+                    {
+                        sb.Append(", ");
+                    }
+                    sb.Append(s);
+                    success = false;
+                }
+            }
+
+            error = (sb.Length == 0) ? string.Empty :
+                $"Namespace: {sb} is/are invalid C# identifier(s)";
+            return success;
+        }
+
+        /// <summary>
+        /// Ensures that the value of <see cref="SubsystemName"/> is valid for the
+        /// C# language.
+        /// </summary>
+        /// <param name="error">String describing the encountered error.</param>
+        /// <returns>True of successful, or false.</returns>
+        public bool ValidateSubsystemName(out string error)
+        {
+            // Ensure a name was provided.
+            if (string.IsNullOrWhiteSpace(SubsystemName))
+            {
+                error = $"Subsystem name: Must specify a name.";
+                return false;
+            }
+
+            bool success = ValidateName(SubsystemName);
+
+            error = success ? string.Empty :
+                $"Subsystem name: {SubsystemName} is not a valid C# identifier.";
+
+            return success;
+        }
+
+        /// <summary>
+        /// Ensures that the necessary templates can be located and successfully loaded.
+        /// </summary>
+        /// <param name="errors">List to which any encountered errors will be added.</param>
+        /// <param name="descriptorTemplate">
+        /// <see cref="FileInfo"/> object representing the subsystem descriptor template file.
+        /// </param>
+        /// <param name="interfaceTemplate">
+        /// <see cref="FileInfo"/> object representing the subsystem interface template file.
+        /// </param>
+        /// <param name="classTemplate">
```

Comment:
I don't think this param exists anymore (@MaxWang-MS, is there any way for the docs gen pipeline to treat something like this as an error?)

Reply:
forgot to rename the param in the doc comment